examples/extensions/update_sitelink_campaign_extension_setting.py

```diff
@@ -0,0 +1,140 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Updates the sitelink campaign extension setting.
+
+Replaces the extension feed items of the sitelink campaign extension setting
+with the given feed item IDs. Note that this doesn't completely remove your
+old extension feed items. See https://developers.google.com/google-ads/api/docs/extensions/extension-settings/overview
+for details.
+"""
+
+
+import argparse
+import sys
+
+from google.api_core import protobuf_helpers
+import google.api_core.path_template
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+from google.ads.google_ads.util import ResourceName
+
+
+def main(client, customer_id, campaign_id, feed_item_ids):
+    """The main method that creates all necessary entities for the example.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+        campaign_id: the campaign ID.
+        feed_item_ids: the IDs the feed items to replace.
+    """
+    campaign_extension_setting_service = client.get_service(
+        "CampaignExtensionSettingService", version="v6"
+    )
+    campaign_extension_setting_operation = client.get_type(
+        "CampaignExtensionSettingOperation", version="v6"
+    )
+    extension_feed_item_service = client.get_service(
+        "ExtensionFeedItemService", version="v6"
+    )
+
+    extension_feed_items = map(
+        lambda feed_item_id: extension_feed_item_service.extension_feed_item_path(
+            customer_id, feed_item_id
+        ),
+        feed_item_ids,
+    )
+    campaign_extension_setting = campaign_extension_setting_operation.update
+    extension_type_enum = client.get_type("ExtensionTypeEnum", version="v6")
+    composite_id = ResourceName.format_composite(
+        # the docs don't make it clear that this should reference the name of the extension type and not the enum
+        # e.g. just extension_type_enum.ExtensionType.SITELINK won't work, but the string 'SITELINK' will
+        campaign_id,
+        extension_type_enum.ExtensionType.Name(
+            extension_type_enum.ExtensionType.SITELINK
+        ),
+    )
+    resource_name = campaign_extension_setting_service.campaign_extension_setting_path(
+        customer_id, composite_id
+    )
+    campaign_extension_setting.resource_name = resource_name
+    # Replace the current extension feed items with the given list
+    campaign_extension_setting.extension_feed_items[:] = extension_feed_items
```

Comment:
Could you move this line underneath the `map` call above so that it groups the logic that deals with this list? Also since the `extension_feed_items` field is empty at first this slicing implies that there might already be items in it. We would normally just call `extend(extension_feed_items)` here, is there a reason that won't work?

Reply:
I thought that since it uses an update mask, extend would add to the existing list rather than replace the items in the list. Is that not the case?

Reply:
Even though this is an update operation, the object you get when calling...

```py
client.get_type("CampaignExtensionSettingOperation", version="v6").update
```

...is a blank instance of `CampaignExtensionOperation` so technically `extend` would _add_ feed items to that repeated field, but we know it's empty because of how the object was created. If the object reference by the resource name attached to the operation has feed items, then the API would decide how to reconcile the existing ones with the list being sent here.

Even if you were working on an instance that was retrieved from the API and had feed items in this list, removing the existing ones and adding new ones wouldn't matter because the API would still reconcile the incoming list with the same logic as when you start with a blank object.

Reply:
Got it, thanks for explaining! I changed it to use extend and confirmed the outcome is the same